restapi/src/main/java/io/stargate/web/docsapi/dao/DocumentDB.java

```diff
@@ -153,14 +158,37 @@ public static boolean containsIllegalChars(String x) {
     return forbiddenCharacters.stream().anyMatch(ch -> x.indexOf(ch) >= 0);
   }
 
-  public static String replaceIllegalChars(String x) {
+  /**
+   * Replaces characters in @param x that are illegal with the hex-encoded equivalent. This makes
+   * the only truly illegal character the percent sign '%'.
+   *
+   * @param x A string with escapable characters, such as `eric@gmail.com`
+   * @return A string with all escapable characters hex-encoded, such as `eric@gmail%2ecom`
+   */
+  public static String replaceEscapableChars(String x) {
     String newStr = x;
-    for (Character y : forbiddenCharacters) {
-      newStr = newStr.replace(y, '_');
+    for (Character y : escapableCharacters) {
+      newStr = newStr.replace(y.toString(), String.format("%%%02x", (int) y));
     }
     return newStr;
   }
 
+  /**
+   * Decodes all hex-encoded characters into their natural form, effectively reversing
+   * DocumentDB#replaceEscapableChars
+   *
+   * @param x A string with encoded characters, such as `eric@gmail%2ecom`
+   * @return A string with all encoded characters decoded, such as `eric@gmail.com`
+   */
+  public static String decodeEscapableChars(String x) {
+    try {
+      return URLDecoder.decode(x, StandardCharsets.UTF_8.name());
```

Comment:
Using `URLDecoder` in this context will probably require the end user to escape `%` signs twice because otherwise the HTTP/Servlet libs are going to interpret them before we get the field name in our code. WDYT?